src/main/java/Deadline.java

```diff
@@ -0,0 +1,33 @@
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    private boolean dateProcessed = true;
+
+    public Deadline(String description, String time) {
+        super(description);
+        this.time = time;
+        try {
+            processedTime = LocalDate.parse(time);
+        } catch (DateTimeException e) {
+            dateProcessed = false;
+        }
+    }
+
+    public String getTaskIcon() {
+        return "D";
+    }
+
+    public boolean hasTime() {
+        return true;
+    }
+
+    @Override
+    public String getTime() {
+        return dateProcessed
+                ? processedTime.format(DateTimeFormatter.ofPattern("MMM d yyyy"))
+                : time;
+    }
+}
```

Comment:
Great implementation! imports are explicitly stated, implementation is clean and the coding style is adhered to.

Reply:
Consider adding a java doc comments to the class for a description of what the class is about.